graphene/relay/connection.py

```diff
@@ -53,7 +53,7 @@ class Meta:
     def __init_subclass_with_meta__(cls, node=None, name=None, **options):
         _meta = ConnectionOptions(cls)
         assert node, "You have to provide a node in {}.Meta".format(cls.__name__)
-        assert issubclass(
+        assert isinstance(node, NonNull) or issubclass(
```

Comment:
Is `NonNull` really the only case where the node is an instance instead of a subclass ? I'm not sure of this :)

Reply:
In theory, we should be able to use instances of `String`, `Int`,`ObjectType` etc. However, because we use this `node` declaration in a `Field` ([line 70](https://github.com/graphql-python/graphene/pull/934/files#diff-50d2eb86a71015fe93ccc0c11414760bR70)), it will throw an exception if we do
```python
class MyConnection(Connection):
    class Meta:
        node = String(required=True)
```

Reply:
Out of that list, `NonNull` is the only one we'll get as an instance.

Reply:
Also, @minhtule 's example of `node = String(required=True)` is invalid as we expect a Type to be passed there. We create the `Node` field internally (as it needs a special name etc).
The only valid non-type input is `node = NonNull(String)`